test/system/post_test.rb

```diff
@@ -122,6 +124,17 @@ def assert_page_does_not_reload(message = "page should not reload")
 
     # Make sure that image has been uploaded
     page.assert_selector('#preview-main img', count: 1)
+
+    find('.publish').click()
+
+    assert_selector('h1', text: "My wiki page")
+    assert_selector('#content', text: "All about this interesting stuff")
+    assert_selector('.alert-success', text: "×\nWiki page created.")
+    expect(page.find('.main-image')['src']).to have_content 'pl.png'
```

Comment:
Is it possible it's order-specific, and that if we (as in the next commit i'll make) put `main-image` class first, it'll work? let's try.

Reply:
Re @Tlazypanda's note:

> So far something like this is detecting fine : page.find('img.rounded', visible: false)['src'])

```suggestion
    expect(page.find('img.main-image')['src']).to have_content 'pl.png'
```
We could try this, but why would it want `visible: false`?

Reply:
OK trying this `img.` addition but it probably won't work:

```

ERROR PostTest#test_drag_and_drop_image_upload_to_wiki_post_editor (268.58s)
Minitest::UnexpectedError:         Capybara::ElementNotFound: Unable to find css ".main-image"
            test/system/post_test.rb:134:in `block in <class:PostTest>'
```

Reply:
Could it be that it's hidden? Why would that be... do we need to wait more?

Reply:
OH WAIT! The client-side display isn't driven by rails `image_tag` -- it's javascript! So we have to add the css class there too! Looking...

Reply:
OK, so it inserts into the same image tag. But, it's hidden until the src is assigned!

https://github.com/publiclab/plots2/blob/1d939f23d9babdfebaa971fb4affc13afce7af0d/app/assets/javascripts/editorToolbar.js#L209-L210

Maybe we're trying to detect too fast, and not waiting until it's shown.

Reply:
Wait thats not right. I think this is for /after/ publishing, so we're not waiting for JS. Sorry.

Could it be the lazy loading of images that we need to wait for?

Reply:
Yes -- here, lazy loading on main images:

https://github.com/publiclab/plots2/pull/8043/files#diff-10d79f07bd4ae39a54ddd901d05832f6ea45f6ec427932edd966bed0b83e6714R7

Let's see if we can work around that... maybe just adding visible: false is OK?

Reply:
OK, now:

```
ERROR PostTest#test_drag_and_drop_image_upload_to_wiki_post_editor (175.89s)
Minitest::UnexpectedError:         Capybara::ElementNotFound: Unable to find css "img.main-image"
            test/system/post_test.rb:134:in `block in <class:PostTest>'

ERROR RichTextEditorTest#test_thumbnail_image_drag_and_drop_upload (400.36s)
Minitest::UnexpectedError:         Capybara::ElementNotFound: Unable to find visible css ".main-image img.d-print-none"
            test/system/rich_text_editor_test.rb:42:in `block in <class:RichTextEditorTest>'
```